mmdet3d/models/backbones/sst.py

```diff
@@ -0,0 +1,472 @@
+        embed_dims(int): Number of dimension of feature embedding.
+        num_heads (int): Number of heads in attention
+        feedforward_channels (int): Number of channels in FFN.
+        attn_dropout(float): Drop probability when do the attention:
+        ffn_dropout(float): Drop probability in the FFN:
+        norm_cfg (:obj:`ConfigDict`): Config for normalization layer.
+        act_cfg (:obj:`ConfigDict`): Config for activation layer.
+        post_norm (bool): Whether use post norm. Defaults to True.
+        init_cfg (:obj:`ConfigDict`): Initialization config.
+            Defaults to None.
+    """
+
+    def __init__(self,
+                 embed_dims,
+                 num_heads,
+                 feedforward_channels=256,
+                 attn_dropout=0.0,
+                 ffn_dropout=0.0,
+                 norm_cfg=dict(type='LN'),
+                 act_cfg=dict(type='GELU'),
+                 post_norm=True,
+                 init_cfg=None):
+
+        super().__init__(init_cfg=init_cfg)
+        self.attn = SparseRegionAttention(embed_dims, num_heads, attn_dropout)
+
+        self.linear1 = nn.Linear(embed_dims, feedforward_channels)
+        self.dropout = nn.Dropout(ffn_dropout)
+        self.linear2 = nn.Linear(feedforward_channels, embed_dims)
+
+        self.norm1 = build_norm_layer(norm_cfg, embed_dims)[1]
+        self.norm2 = build_norm_layer(norm_cfg, embed_dims)[1]
+
+        self.dropout1 = nn.Dropout(ffn_dropout)
+        self.dropout2 = nn.Dropout(ffn_dropout)
+
+        self.activation = build_activation_layer(act_cfg)
+        self.post_norm = post_norm
+        self.fp16_enabled = False
+
+    @auto_fp16(apply_to=('feat'))
+    def forward(
+        self,
+        feat,
+        batching_position_embed,
+        batching_padding_mask,
+        seq_win_mappings,
+    ):
+        """
+        Args:
+            feat (tensor): The sequence features to be transformed.
+                has shape (N, FEAT_DIM).
+            batching_position_embed (list[tensor]): The list indicate
+                the different batching. Each tensor is the position
+                embedding of corresponding batching.
+                Each tensor has shape (NUM_WINS, NUM_TOKEN, FEAT_DIM).
+            batching_padding_mask (list[tensor]): The  list
+                indicate the different batching. The tensor is
+                the padding mask of corresponding batching, each
+                tensor has shape (NUM_WINS, NUM_TOKEN).
+            seq_win_mappings (list[tuple]): Each tuple contains
+            three important information of corresponding
+            batching.
+
+                - A identical index of voxel in corresponding
+                  batching windows
+                - The index of voxel in original sequence.
+                - The number of token of each window in this batching.
+
+        Returns:
+            tensor: The sequence features after attention.
+            Has shape (N, FEAT_DIM).
```

Comment:
Need indent?

Reply:
There is a special syntax for `Returns` in sphinx